extensions/amp-analytics/0.1/crypto-impl.js

```diff
@@ -20,29 +20,62 @@ import {getService} from '../../../src/service';
 export class Crypto {
 
   constructor(win) {
-    this.win = win;
+    if (win.crypto) {
+      this.subtle = win.crypto.subtle || win.crypto.webkitSubtle;
+    }
   }
 
   /**
-   * Returns the SHA-384 hash of the input string in an ArrayBuffer.
+   * Returns the SHA-384 hash of the input string in a number array.
+   * Input string cannot contain chars out of range [0,255].
    * @param {string} str
-   * @returns {!Promise<!ArrayBuffer>}
+   * @returns {!Promise<!Array<number>>}
+   * @throws {!Error} when input string contains chars out of range [0,255]
    */
   sha384(str) {
+    if (this.subtle) {
+      try {
+        return this.subtle.digest('SHA-384', str2ab(str))
+            // [].slice.call(Unit8Array) is a shim for Array.from(Unit8Array)
+            .then(buffer => [].slice.call(new Uint8Array(buffer)),
+                () => lib.sha384(str));
+      } catch (e) {
+        // ignore, fallback to closure lib.
```

Comment:
Potentially any browser doesn't follow the standard API can throw.
For example, 

- Chrome 37 throws because it has a different method signature. 
- IE11 doesn't return `Promise` (though already excluded as it's in window.msCrypto)